suite-common/trading/src/thunks/exchange/sendDexTransactionThunk.ts

```diff
@@ -0,0 +1,119 @@
+import { ExchangeTrade } from 'invity-api';
+
+import { createThunk } from '@suite-common/redux-utils';
+import { Account } from '@suite-common/wallet-types';
+
+import { exchangeThunks, tradingThunks } from '../';
+import { TRADING_EXCHANGE_THUNK_PREFIX } from '../../constants';
+import { tradingActions } from '../../reducers/tradingReducer';
+import { selectTradingExchangeSelectedQuote } from '../../selectors/tradingSelectors';
+import { RecomposeAndSignTxThunkProps } from '../common/recomposeAndSignTxThunk';
+
+export type SendDexTransactionThunkProps = {
+    account: Account;
+    returnUrl: string;
+    setMaxOutputId?: number | undefined;
+
+    nextStep: () => void;
+    processResponseData: (response: ExchangeTrade) => void;
+    triggerAnalyticsTradeConfirmation: () => void;
+    signAndPushSendFormTransaction: RecomposeAndSignTxThunkProps['signAndPushSendFormTransaction'];
+};
+
+export const sendDexTransactionThunk = createThunk(
+    `${TRADING_EXCHANGE_THUNK_PREFIX}/sendDexTransaction`,
+    async (
+        {
+            account,
+            returnUrl,
+            setMaxOutputId,
+            nextStep,
+            processResponseData,
+            triggerAnalyticsTradeConfirmation,
+            signAndPushSendFormTransaction,
+        }: SendDexTransactionThunkProps,
+        { dispatch, getState, rejectWithValue },
+    ) => {
+        const selectedQuote = selectTradingExchangeSelectedQuote(getState());
+
+        if (
+            !selectedQuote ||
+            !selectedQuote.dexTx ||
+            !selectedQuote.receiveAddress ||
+            (selectedQuote.status !== 'APPROVAL_REQ' && selectedQuote.status !== 'CONFIRM')
```

Comment:
I have a small dilemma, isn't it more readable?
```suggestion
            !['APPROVAL_REQ', 'CONFIRM'].includes(selectedQuote.status)
```

Reply:
I should also control whether `status` is undefined, so I will let it as it is, ok? :)

Reply:
Updated

Reply:
Sure, that makes sense to me.